### DISCUSSION TITLE
while building i got this output....

### DISCUSSION THREAD

**Discussion by jyotrimaya1993** [OpenFAST/openfast#1126] [Q&A]
while building i got this output....
why simulink is skipped and why some warning kindly tell me

```
Build started...
1>------ Build started: Project: FAST_Registry, Configuration: Release Win32 ------
1>data.c
1>gen_c_types.c
1>gen_module_files.c
1>misc.c
1>my_strtok.c
1>registry.c
1>reg_parse.c
1>sym.c
1>symtab_gen.c
1>type.c
1>Generating code
1>0 of 75 functions ( 0.0%) were compiled, the rest were copied from previous compilation.
1>  0 functions were new in current compilation
1>  0 functions had inline decision re-evaluated but remain unchanged
1>Finished generating code
1>FAST_Registry.vcxproj -> C:\openfast\openfast-main\build\bin\Registry.exe
2>------ Build started: Project: MAP_dll, Configuration: Debug x64 ------
2>bstraux.c
2>C:\openfast\openfast-main\modules\map\src\bstring\bstraux.c(1004,26): warning C4267: 'function': conversion from 'size_t' to 'int', possible loss of data
2>C:\openfast\openfast-main\modules\map\src\bstring\bstraux.c(1017,13): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
2>bstrlib.c
2>C:\openfast\openfast-main\modules\map\src\bstring\bstrlib.c(367,32): warning C4267: 'function': conversion from 'size_t' to 'int', possible loss of data
2>dpmpar.c
2>enorm.c
2>enorm_u.c
2>lmder.c
2>lmpar.c
2>qrfac.c
2>qrsolv.c
2>freedata.c
2>jacobian.c
2>lineroutines.c
2>mapapi.c
2>maperror.c
2>mapinit.c
2>numeric.c
2>outputstream.c
2>residual.c
2>simclist.c
2>Generating Code...
2>lmroutines.cc
2>MAP_dll.vcxproj -> C:\openfast\openfast-main\build\bin\MAP_x64.lib
2>Done building project "MAP_dll.vcxproj".
C:\openfast\openfast-main\modules\nwtc-library\src\SysIVF.f90 : warning: Module 'x64\Debug\syssubs.mod' is created by both 'C:\openfast\openfast-main\modules\nwtc-library\src\SysIVF.f90' and 'C:\openfast\openfast-main\modules\nwtc-library\src\SysMatlabWindows.f90'.
3>------ Build started: Project: FASTlib (IFORT), Configuration: Debug x64 ------
3>Performing Pre-Build Event...
3>'git' is not recognized as an internal or external command,
3>operable program or batch file.
3>'git' is not recognized as an internal or external command,
3>operable program or batch file.
3>Creating library...
3>
3>Build log written to  "file://C:\openfast\openfast-main\vs-build\FASTlib\x64\Debug\BuildLog.htm"
3>FASTlib - 0 error(s), 0 warning(s)
4>------ Skipped Build: Project: OpenFAST-Simulink (IFORT), Configuration: Debug_Matlab x64 ------
4>Project not selected to build for this solution configuration 
5>------ Build started: Project: FAST (IFORT), Configuration: Debug x64 ------
5>Linking...
5>FASTlib_x64_Debug.lib(mapapi.obj) : warning LNK4075: ignoring '/EDITANDCONTINUE' due to '/INCREMENTAL:NO' specification
5>
5>Build log written to  "file://C:\openfast\openfast-main\vs-build\FAST\x64\Debug\BuildLog.htm"
5>FAST - 0 error(s), 1 warning(s)
========== Build: 4 succeeded, 0 failed, 0 up-to-date, 1 skipped ==========
```

_Originally posted by @jyotrimaya1993 in https://github.com/OpenFAST/openfast/issues/1098#issuecomment-1123452223_

**Comment by jyotrimaya1993** [comment]
![2](https://user-images.githubusercontent.com/104576729/167823593-428c4dca-6d8d-4007-9ce6-b4a5130d124f.jpg)

![1](https://user-images.githubusercontent.com/104576729/167823636-3d182b48-3933-490a-9d0a-17c7df6a5f7a.jpg)
these are the errors i got......is it bcz of math library?

**Reply by rafmudaf** [reply]
Also please change the title of this discussion to reflect your question so that other users can find the answers here easier.

**Reply by bjonkman** [reply]
These are actually warnings, not errors. Many of them are warning that the code is non-standard Fortran 2003 (almost all of those warnings are from code using the Fortran 2008 standard instead). The ones from the "CUSTOMBUILD" are just warnings about how the FAST Registry creates c code for the data types that are used. Developers should be taking a look at the list of warnings and determining if they are important or not. In your case, you can ignore these.

**Comment by rafmudaf** [comment]
Which version of OpenFAST are you trying to build?

**Reply by jyotrimaya1993** [reply]
3.1.0 version

**Comment by bjonkman** [comment]
When you build any of the configurations, the Visual Studio solution file has been set up to know which projects are needed for that particular build configuration. For a "normal" OpenFAST executable, (Release, Debug, Release_Double, or Debug_Double configurations), it builds the `FAST` project, but not the `OpenFAST-Simulink` project. If you choose to build a DLL for Matlab (Release_Matlab or Debug_Matlab configurations), it will build the `OpenFAST-Simulink` project but not the `FAST` project. It can't build both the .exe and .dll at the same time because they require slightly different compilation options. 

You chose the `Debug | x64` configuration, so it knows not to build the `OpenFAST-Simulink` dll:
```
4>------ Skipped Build: Project: OpenFAST-Simulink (IFORT), Configuration: Debug_Matlab x64 ------
4>Project not selected to build for this solution configuration
```

**Comment by navkum1099** [comment]
So, how these errors can be ignored and configuration can be achieved? 
I think we've to comment '/INCREMENTAL:NO' specification but I don't know in which file of OpenFASTI'll get it. Please let me know. Thanks.
